handlers/update_task.go

```diff
@@ -0,0 +1,46 @@
+package handlers
+
+import (
+	domainErr "github.com/AliShan302/Task_Schedular/errors"
+	"time"
+
+	"github.com/go-openapi/runtime/middleware"
+
+	runtime "github.com/AliShan302/Task_Schedular"
+	"github.com/AliShan302/Task_Schedular/gen/restapi/operations"
+	"github.com/AliShan302/Task_Schedular/models"
+)
+
+// NewUpdateTask handles request for updating task
+func NewUpdateTask(rt *runtime.Runtime) operations.UpdateTaskHandler {
+	return &updateTask{rt: rt}
+}
+
+type updateTask struct {
+	rt *runtime.Runtime
+}
+
+// Handle the update task request
+func (d *updateTask) Handle(params operations.UpdateTaskParams) middleware.Responder {
+	task := models.Task{
+		ID:          params.Task.ID,
+		Name:        params.Task.Name,
+		CreatedAt:   time.Time(params.Task.CreatedAt),
+		Status:      params.Task.Status,
+		Description: params.Task.Description,
+		Deadline:    time.Time(params.Task.Deadline),
+		Priority:    int(params.Task.Priority),
+		Assignee:    params.Task.Assignee,
+	}
+	if err := d.rt.Service().UpdateTask(&task); err != nil {
+		switch apiErr := err.(*domainErr.APIError); {
+		case apiErr.IsError(domainErr.NotFound):
+			return operations.NewUpdateTaskNotFound()
```

Comment:
You are not returning 404 anywhere in the UpdateTask request so I don't think this case will ever be true.
You need to add some logic in the service layer and return relevant errors accordingly.